packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -1128,6 +1128,11 @@ export class CodeModelBuilder {
               explode = true;
               break;
           }
+
+          if (param.explode && !param.collectionFormat) {
+            style = SerializationStyle.Form;
+            explode = true;
+          }
```

Comment:
Handle TCGC `explode=true, collectionFormat=undefined` on
```
      @route("array{?param*}")
      op array(param: string[]): void;
```
<-- I am not sure what this intend to map, will check with Chenjie when he's back.

But `collectionFormat` exists in
```
@route("array")
op array(
  @query(#{ explode: true })
  param: string[],
): void;
```